integrations/cdi/jpa-cdi/src/test/java/io/helidon/integrations/cdi/jpa/chirp2/TestCascadePersist2.java

```diff
@@ -0,0 +1,302 @@
+import jakarta.enterprise.context.BeforeDestroyed;
+import jakarta.enterprise.event.Observes;
+import jakarta.enterprise.inject.se.SeContainer;
+import jakarta.enterprise.inject.se.SeContainerInitializer;
+import jakarta.inject.Inject;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.PersistenceContext;
+import jakarta.persistence.PersistenceContextType;
+import jakarta.persistence.SynchronizationType;
+import jakarta.transaction.HeuristicMixedException;
+import jakarta.transaction.HeuristicRollbackException;
+import jakarta.transaction.NotSupportedException;
+import jakarta.transaction.RollbackException;
+import jakarta.transaction.Status;
+import jakarta.transaction.SystemException;
+import jakarta.transaction.TransactionManager;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.CoreMatchers.sameInstance;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.fail;
+
+@ApplicationScoped
+@DataSourceDefinition(
+    name = "chirp2",
+    className = "org.h2.jdbcx.JdbcDataSource",
+    url = "jdbc:h2:mem:TestCascadePersist2;DB_CLOSE_DELAY=-1;MODE=LEGACY;INIT=SET TRACE_LEVEL_FILE=4\\;RUNSCRIPT FROM 'classpath:chirp2.ddl'\\;",
+    serverName = "",
+    properties = {
+        "user=sa"
+    }
+)
+class TestCascadePersist2 {
+
+    private SeContainer cdiContainer;
+
+    @Inject
+    private TransactionManager transactionManager;
+
+    @PersistenceContext(
+        type = PersistenceContextType.TRANSACTION,
+        synchronization = SynchronizationType.SYNCHRONIZED,
+        unitName = "chirp2"
+    )
+    private EntityManager em;
+
+
+    /*
+     * Constructors.
+     */
+
+
+    TestCascadePersist2() {
+        super();
+    }
+
+
+    /*
+     * Setup and teardown methods.
+     */
+
+
+    @BeforeEach
+    void startCdiContainerAndRunDDL() throws SQLException {
+        System.setProperty(io.helidon.integrations.cdi.jpa.JpaExtension.class.getName() + ".enabled", "false");
+        System.setProperty(io.helidon.integrations.cdi.jpa.PersistenceExtension.class.getName() + ".enabled", "true");
+        final SeContainerInitializer initializer = SeContainerInitializer.newInstance()
+            .addBeanClasses(this.getClass());
+        this.cdiContainer = initializer.initialize();
+        /*
+        final DataSource ds = this.cdiContainer.select(DataSource.class).get();
+        assertThat(ds, notNullValue());
+        try (final Connection connection = ds.getConnection();
+             final Statement statement = connection.createStatement()) {
+            assertThat(statement, notNullValue());
+            statement.executeUpdate("RUNSCRIPT FROM 'classpath:chirp2.ddl'");
+        }
+        */
```

Comment:
Maybe uncomment (if we want these lines) or remove (if we don't want them)?

Reply:
Derp; good catch; thanks